tools/pythonpkg/pyduckdb/filesystem.py

```diff
@@ -8,6 +8,15 @@ class ModifiedMemoryFileSystem(MemoryFileSystem):
 	# defer to the original implementation that doesn't hardcode the protocol
 	_strip_protocol = classmethod(AbstractFileSystem._strip_protocol.__func__)
 
+	# Add this manually because it's apparently missing on windows???
```

Comment:
How can this be missing on windows? It's part of the base abstract class?

Reply:
<https://github.com/Tishj/duckdb/actions/runs/4330367153/jobs/7561664571>
```
  ================================== FAILURES ===================================
  _______________________ TestReadCSV.test_read_filelike ________________________
  
  self = <test_read_csv.TestReadCSV object at 0x06DEB550>
  duckdb_cursor = <duckdb.DuckDBPyConnection object at 0x0744AC80>
  
      def test_read_filelike(self, duckdb_cursor):
      	string = StringIO("c1,c2,c3\na,b,c")
  >   	res = duckdb_cursor.read_csv(string, header=True).fetchall()
  E    AttributeError: 'ModifiedMemoryFileSystem' object has no attribute 'unstrip_protocol'
```

Reply:
I'm not going to claim to be able to understand why this was necessary, but it definitely was

Reply:
That is bizarre. Something to follow up on later maybe, when we add file object support to read_json? ;)

Reply:
That should hopefully only need a small change 🤞

Reply:
Following up on this:
Looking at <https://filesystem-spec.readthedocs.io/en/latest/_modules/fsspec/implementations/memory.html> it doesn't look like it implements `unstrip_protocol`

Reply:
Also this post <https://github.com/cython/cython/issues/3948> *might* be relevant. We don't explicitly initialize the base with `super`

Reply:
Okay, I tried it out on it's own in a standalone workflow, and it works fine? 

https://github.com/Mause/dummy-duckdb-project-5161/actions/runs/4352222833/jobs/7604742821
and
https://github.com/Mause/dummy-duckdb-project-5161/blob/main/windows.py

Maybe it's related to the python version and fsspec version?

Reply:
Okay, reproduced the failure on Python 3.6? https://github.com/Mause/dummy-duckdb-project-5161/actions/runs/4352254997/jobs/7604808575